pkg/tests/tasks/observability/prometheus_extension_provider_test.go

```diff
@@ -0,0 +1,497 @@
+metadata:
+  name: custom-prometheus-operators
+spec:
+  targetNamespaces:
+    - %s`,
+			ns),
+		`
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: rhods-prometheus-operator
+spec:
+  channel: beta
+  name: rhods-prometheus-operator
+  source: redhat-operators 
+  sourceNamespace: openshift-marketplace`)
+}
+
+func createSmcpWithPrometheusExtensionProvider(t test.TestHelper, smcpNs, prometheusNs, additionalSmmrNs string) {
+	t.T().Helper()
+	oc.ApplyString(t, smcpNs, `
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: basic
+spec:
+  addons:
+    grafana:
+      enabled: false
+    kiali:
+      enabled: false
+    prometheus:
+      enabled: false
+  extensionProviders:
+  - name: prometheus
+    prometheus: {}
+  gateways:
+    egress:
+      enabled: false
+    openshiftRoute:
+      enabled: false
+  proxy:
+    accessLogging:
+      file:
+        name: /dev/stdout
+  security:
+    dataPlane:
+      mtls: true
+  tracing:
+    type: None`,
+		fmt.Sprintf(`
+apiVersion: maistra.io/v1
+kind: ServiceMeshMemberRoll
+metadata:
+  name: default
+spec:
+  members:
+  - %s
+  - %s`,
+			prometheusNs,
+			additionalSmmrNs))
+}
+
+func installPrometheus(t test.TestHelper, ns string, permittedNs ...string) {
+	t.T().Helper()
+	oc.ApplyString(t, ns,
+		fmt.Sprintf(`
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: prometheus
+spec:
+  securityContext: {}
+  serviceAccountName: prometheus-k8s
+  podMonitorSelector: {}
+  podMonitorNamespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: %s
+  serviceMonitorSelector: {}
+  serviceMonitorNamespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: %s
+  podMetadata:
+    annotations:
+      sidecar.istio.io/inject: "true"
+      traffic.sidecar.istio.io/includeInboundPorts: ""
+      traffic.sidecar.istio.io/includeOutboundIPRanges: ""
+      proxy.istio.io/config: |
+        proxyMetadata:
+          OUTPUT_CERTS: /etc/istio-output-certs
+      sidecar.istio.io/userVolumeMount: '[{"name": "istio-certs", "mountPath": "/etc/istio-output-certs"}]'
+  volumes:
+  - name: istio-certs
+    emptyDir:
+      medium: Memory
+  volumeMounts:
+  - mountPath: /etc/prom-certs/
+    name: istio-certs`,
+			ns,
+			ns))
+
+	for _, permitNs := range permittedNs {
+		oc.ApplyString(t, permitNs, `
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: custom-prometheus-permissions
+rules:
+- apiGroups: [""]
+  resources:
+  - services
+  - endpoints
+  - pods
+  verbs: ["get", "list", "watch"]
+- apiGroups: [""]
+  resources:
+  - configmaps
+  verbs: ["get"]`, `
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: custom-prometheus-permissions
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: custom-prometheus-permissions
+subjects:
+- kind: ServiceAccount
+  name: prometheus-k8s
+  namespace: custom-prometheus`)
+	}
+}
```

Comment:
I think you don't need these roles when you run tests as cluster admin. Did you try to remove them?

Reply:
This causes a lot of "<pods/services/endpoints> is forbidden" messages in the logs of the Prometheus pod which reference the bookinfo and istio-system namespaces. None of the errors referenced the custom-prometheus namespace, however, so I was able to remove that instance of the Role+RoleBinding.